### ISSUE TITLE
can't parse a domain with underscore

### ISSUE THREAD

**Issue by vzaidman** [medialize/URI.js#347]
![banner](https://user-images.githubusercontent.com/5188459/29117829-624bf2a0-7d08-11e7-8e34-36d6367eb2ee.png)

probably because of this code
```
  // allowed hostname characters according to RFC 3986
  // ALPHA DIGIT "-" "." "_" "~" "!" "$" "&" "'" "(" ")" "*" "+" "," ";" "=" %encoded
  // I've never seen a (non-IDN) hostname other than: ALPHA DIGIT . -
  URI.invalid_hostname_characters = /[^a-zA-Z0-9\.-]/;
```

**Comment by rodneyrehm** [comment]
Do you want to send a PR for that? (maybe also adding an URL with a _ in its domain to [urls.js](https://github.com/medialize/URI.js/blob/gh-pages/test/urls.js) for testing)

**Comment by vzaidman** [comment]
sure ill send a pr

**Comment by vzaidman** [comment]
submitted a pr:
https://github.com/medialize/URI.js/pull/349

**Comment by vzaidman** [comment]
submitted a PR with a relevant test:
https://github.com/medialize/URI.js/pull/348

**Comment by vzaidman** [comment]
@rodneyrehm is there a possibility to have 
```
host:name.something.com
```
?

**Comment by rodneyrehm** [comment]
the hostname could be an IPv6 address and that may contain `:`. A domain name may not - as far as I know - but then, I wasn't aware `_` was legal either…

**Comment by vzaidman** [comment]
host:name.something.com currently fails in tests because name.something.com is considered the port

**Comment by rodneyrehm** [comment]
When you opened the PR I thought I'd immediately close the issue because `_` is not legal in domain names per [RFC 1034](https://tools.ietf.org/html/rfc1034#section-3.5). However, the [URL spec](https://url.spec.whatwg.org/) seems to permit it and Chrome's implementation parses it. That said, `:` is considered an [illegal code point](https://url.spec.whatwg.org/#host-miscellaneous)

**Comment by vzaidman** [comment]
gotcha. so the PR can be merged i think.

**Comment by rodneyrehm** [comment]
fixed in [v1.18.12](https://github.com/medialize/URI.js/releases/tag/v1.18.12)
